roll/exceptions.py

```diff
@@ -0,0 +1,171 @@
+from utils.exceptions import InternalError, OutputTooLargeError, WarningError
+
+
+def trace2log(trace):
+    trace = trace[1:]  # The first element is always an empty let statement
+    out = "Exception in\n    " + "\nin\n    ".join([str(t) for t in reversed(trace)])
+    return out
+
+
+"""User errors"""
+
+
+class RunTimeError(Exception):
+    """Raised when an error occurs while evaluating a program"""

```

Comment:
It's customary to add a `pass` to empty code, even with docstrings.
```suggestion
class RunTimeError(Exception):
    """Raised when an error occurs while evaluating a program"""
    pass
```